rabbus/lib/topology/index.js

```diff
@@ -0,0 +1,126 @@
+var EventEmitter = require("events").EventEmitter;
+var util = require("util");
+
+var logger = require("../logging")("rabbus.topology");
+var OptionParser = require("../optionParser");
+
+var startKey = Symbol("started");
+
+// Topology
+// --------
+
+function Topology(rabbit, options, defaults){
+  this.rabbit = rabbit;
+  this.options = OptionParser.parse(options, defaults);
+
+  this.routingKey = this.options.routingKey;
+  this.messageType = this.options.messageType;
+  this.exchange = this.options.exchange;
+  this.queue = this.options.queue;
+  this.routingKey = this.options.routingKey;
+}
+
+// Type Members
+// ------------
+
+Topology.verify = function(rabbit, options, defaults, cb){
+  var topology;
+
+  if (isTopology(options)){
+    topology = options;
+    cb(undefined, topology);
+  } else {
+    topology = new Topology(rabbit, options, defaults);
+    topology.execute((err) => {
+      return cb(err, topology);
+    });
+  }
+};
+
+// Public Members
+// --------------
+
+Topology.prototype.execute = function(cb){
```

Comment:
yeah, i went back and forth on this for a while. init, execute, create, run, configure, doIt, someMethodThatRunsThings, turnThisIntoARealTopology :P

i think i like init better than execute... i thought that it might be confusing, but now i can't remember why i thought that when i was naming this method